frb_rust/src/handler.rs

```diff
@@ -241,51 +247,44 @@ impl<EH: ErrorHandler> Executor for ThreadPoolExecutor<EH> {
                         }
                     }
                     Err(error) => {
-                        eh2.handle_error(wrap_info2.port.unwrap(), Error::ResultError(error));
+                        eh2.handle_error(wrap_info2.port.unwrap(), Error::CustomError(Box::new(error)));
                     }
                 };
             });
 
             if let Err(error) = thread_result {
-                eh.handle_error(wrap_info.port.unwrap(), Error::Panic(error));
+                eh.handle_panic(wrap_info.port.unwrap(), Error::Panic(error));
             }
         });
     }
 
-    fn execute_sync<SyncTaskFn>(
+    fn execute_sync<SyncTaskFn, Er>(
         &self,
         _wrap_info: WrapInfo,
         sync_task: SyncTaskFn,
-    ) -> Result<SyncReturn<Vec<u8>>>
+    ) -> Result<SyncReturn<Vec<u8>>, Er>
     where
-        SyncTaskFn: FnOnce() -> Result<SyncReturn<Vec<u8>>> + UnwindSafe,
+        SyncTaskFn: FnOnce() -> Result<SyncReturn<Vec<u8>>, Er> + UnwindSafe,
+        Er: IntoDart,
     {
         sync_task()
     }
 }
 
 /// Errors that occur from normal code execution.
-#[derive(Debug)]
+//#[derive(Debug)]
 pub enum Error {
-    /// Errors from an [anyhow::Error].
-    ResultError(anyhow::Error),
+    /// Errors that implement [IntoDart].
+    CustomError(Box<dyn BoxIntoDart>),
     /// Exceptional errors from panicking.
     Panic(Box<dyn Any + Send>),
 }
 
 impl Error {
-    /// The identifier of the type of error.
-    pub fn code(&self) -> &'static str {
-        match self {
-            Error::ResultError(_) => "RESULT_ERROR",
-            Error::Panic(_) => "PANIC_ERROR",
-        }
-    }
-
     /// The message of the error.
     pub fn message(&self) -> String {
         match self {
-            Error::ResultError(e) => format!("{:?}", e),
+            Error::CustomError(_e) => "custom error".to_string(),
```

Comment:
Shall we use the old code, i.e. `Error::ResultError(e) => format!("{:?}", e)` which reveal more information

Reply:
Box<dyn IntoDart> does not implmement debug

Reply:
neither dyn IntoDart and etc

Reply:
I see. But for anyhow errors, it does implement debug and prints stack trace (which is very useful!).

Shall we do sth like this:

```
trait ErrorMessage {
  fn message(&self) -> Option<String>;
}

impl MaybeDebug for anyhow::Error { fn message() { format!("{:?}", self) } }

// blanket
impl<T> MaybeDebug for T { fn message() { None } }